File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.swift
import Foundation

func solution(_ arr:[[Int]]) -> [[Int]] {
    
    var result = arr
    
    if result.count > result[0].count {
        for i in 0..<(result.count - result[0].count) {
            for j in 0..<result.count { result[j].append(0) }
        }
        return result
    } else if result.count < result[0].count {
        for i in 0..<(result[0].count - result.count) {
            result.append(Array(repeating: 0, count: result[0].count))
        }
        return result
    } else {
        return result
    }
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr) {
    vector<vector<int>> answer;

    if (arr.size() > arr[0].size()) {
        int diff = arr.size() - arr[0].size();
        for (vector<int> &v : arr) {
            for (int i = 0; i < diff; i++) {
                v.push_back(0);
            }
        }
    } else if (arr.size() < arr[0].size()) {
        int diff = arr[0].size() - arr.size();
        int size = arr[0].size();
        for (int i = 0; i < diff; i++) {
            vector<int> v(size, 0);
            arr.push_back(v);
        }
    } else {
        return arr;
    }

    return arr;
}